packages/vite/src/node/build.ts

```diff
@@ -417,7 +418,7 @@ async function doBuild(
         entryFileNames: ssr
           ? `[name].js`
           : libOptions
-          ? `${libOptions.fileName || pkgName}.${output.format || `es`}.js`
+          ? resolveLibFilename(libOptions, output.format, pkgName)
```

Comment:
When `output.format` is `undefined` and `output.fileName` is a function, it should still be properly called.

I think the cleanest is to keep the default for output.format here:
```suggestion
          ? resolveLibFilename(libOptions, output.format || 'es', pkgName)
```